src/vs/workbench/contrib/chat/browser/chatEditing/chatEditingSession.ts

```diff
@@ -303,11 +299,17 @@ export class ChatEditingSession extends Disposable implements IChatEditingSessio
 			}
 		}
 
-		const entriesArr: ChatEditingModifiedFileEntry[] = [];
+		const entriesArr: IModifiedFileEntry[] = [];
 		// Restore all entries from the snapshot
 		for (const snapshotEntry of snapshot.entries.values()) {
 			const entry = await this._getOrCreateModifiedFileEntry(snapshotEntry.resource, snapshotEntry.telemetryInfo);
-			entry.restoreFromSnapshot(snapshotEntry);
+			if (entry.kind === 'text' && snapshotEntry.kind === 'text') {
+				entry.restoreFromSnapshot(snapshotEntry);
+			} else if (entry.kind === 'notebook' && snapshotEntry.kind === 'notebook') {
+				throw new Error('Not implemented');
```

Comment:
We now have a way to capture/restore snapshots for notebooks
And given we have interfaces for each, we can have separate concrete implementations for each